File: case02/python/Produto.py
from Data import Data

class Produto:
    def __init__(self,nome:str,preco:float,data:Data):
        self.nome = nome
        self.preco = preco
        self.data = data
    
    def get_nome(self):
        return self.nome
    
    def get_preco(self):
        return self.preco
    
    def get_data(self):
        return self.data
    
    def set_nome(self, nome):
        self.nome = nome
    
    def set_preco(self, preco):
        self.preco = preco

    def set_data(self, data):
        self.data = data

    def __str__(self):
        return f"Produto {self.nome}, R$ {self.preco}, Validade: {self.data}"
    
    def verificaProdutoVencido(self,hoje:Data):
        difDia = hoje.get_dia() - self.data.get_dia()
        difMes = hoje.get_mes() - self.data.get_mes()
        difAno = hoje.get_ano() - self.data.get_ano()
        if(difAno > 0):
            return False
        elif (difAno < 0):
            return True
        else:
            if(difMes > 0):
                return False
            elif(difMes < 0):
                return True
            else:
                if(difDia >= 0):
                    return False
                else:
                    return True

                

File: case02/java/Produto.java
public class Produto
{
  private String nome;
  private double preco;
  private Data data;
  
  public Produto(String nome, double preco, Data data){
      this.nome = nome;
      this.preco = preco;
      this.data = data;
  }
  public String getNome(){
      return nome;
  }
  public double getPreco(){
      return preco;
  }
  public Data getData(){
      return data;
  }
  public void setNome(String nome){
      this.nome = nome;
  }
  public void setPreco(double preco){
      this.nome = nome;
  }
  public void setData(Data data){
      this.data = data;
  }
  public String toString(){
      return "Produto: " +nome+ " R$: " +preco+ 
      "\nValidade: " +data;     
  }
  public boolean verificaProdutoVencido(Data hoje){
    int difDia = hoje.getDia() -  data.getDia();
    int difMes = hoje.getMes() -  data.getMes();
    int difAno = hoje.getAno() -  data.getAno();
    if (difAno > 0){
        return false;
    }else if (difAno < 0){
        return true;
    }else{
        if (difMes > 0){
            return false;
        }
        else if (difMes <0){
            return true;
        }else{
           if (difDia >= 0){
               return false; 
           }else{
            return true;
        }
     }
    }
}
}

